puppet/lib/puppet/parser/functions/query.rb

```diff
@@ -0,0 +1,45 @@
+Puppet::Parser::Functions.newfunction(:query, :type => :rvalue, :arity => -3, :doc => <<-EOT
```

Comment:
Hmm. Acceptance test perhaps? With the function in a manifest? I can't think of any other way to test this without changing up what we do in some majorish way. Hrm.